java/src/test/java/org/apache/arrow/gandiva/evaluator/BaseNativeEvaluatorTest.java

```diff
@@ -0,0 +1,238 @@
+  protected ArrowType int32;
+  protected ArrowType int64;
+  protected ArrowType float64;
+
+  @Before
+  public void init() {
+    allocator = new RootAllocator(Long.MAX_VALUE);
+    boolType = new ArrowType.Bool();
+    int8 = new ArrowType.Int(8, true);
+    int32 = new ArrowType.Int(32, true);
+    int64 = new ArrowType.Int(64, true);
+    float64 = new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE);
+  }
+
+  ArrowBuf buf(byte[] bytes) {
+    ArrowBuf buffer = allocator.buffer(bytes.length);
+    buffer.writeBytes(bytes);
+    return buffer;
+  }
+
+  ArrowBuf arrowBufWithAllValid(int size) {
+    int bufLen = (size + 7) / 8;
+    ArrowBuf buffer = allocator.buffer(bufLen);
+    for(int i = 0; i < bufLen; i++) {
+      buffer.writeByte(255);
+    }
+
+    return buffer;
+  }
+
+  ArrowBuf intBuf(int[] ints) {
+    ArrowBuf buffer = allocator.buffer(ints.length * 4);
+    for (int i = 0; i < ints.length; i++) {
+      buffer.writeInt(ints[i]);
+    }
+    return buffer;
+  }
+
+  ArrowBuf longBuf(long[] longs) {
+    ArrowBuf buffer = allocator.buffer(longs.length * 8);
+    for (int i = 0; i < longs.length; i++) {
+      buffer.writeLong(longs[i]);
+    }
+    return buffer;
+  }
+
+  ArrowBuf doubleBuf(double[] data) {
+    ArrowBuf buffer = allocator.buffer(data.length * 8);
+    for (int i = 0; i < data.length; i++) {
+      buffer.writeDouble(data[i]);
+    }
+
+    return buffer;
+  }
+
+  ArrowBuf stringToMillis(String[] dates) {
+    ArrowBuf buffer = allocator.buffer(dates.length * 8);
+    for(int i = 0; i < dates.length; i++) {
+      Instant instant = Instant.parse(dates[i]);
+      buffer.writeLong(instant.getMillis());
+    }
+
+    return buffer;
+  }
+
+  void generateData(DataAndVectorGenerator generator, int numRecords, ArrowBuf buffer) {
+    for(int i = 0; i < numRecords; i++) {
+      generator.writeData(buffer);
+    }
+  }
+
+  private long generateDataAndEvaluate(DataAndVectorGenerator generator,
+                                       NativeEvaluator evaluator,
+                                       int numFields, int numExprs,
+                                       int numRows, int maxRowsInBatch,
+                                       int inputFieldSize)
+    throws GandivaException, Exception {
+    int numRemaining = numRows;
+    List<ArrowBuf> inputData = new ArrayList<ArrowBuf>();
+    List<ArrowBuf> dataBufs = new ArrayList<ArrowBuf>();
+    List<ArrowFieldNode> fieldNodes = new ArrayList<ArrowFieldNode>();
+    List<ValueVector> outputVectors = new ArrayList<ValueVector>();
+
+    long start;
+    long finish;
+    long elapsedTime = 0;
+
+    // set up output vectors
+    // for each expression, generate the output vector
+    for (int i = 0; i < numExprs; i++) {
+      ValueVector valueVector = generator.generateOutputVector(maxRowsInBatch);
+      outputVectors.add(valueVector);
+    }
+
+    // set the bitmap
+    ArrowBuf validity = arrowBufWithAllValid(maxRowsInBatch);
```

Comment:
Where are we releasing this buffer? When the test finishes and the allocator is closed, I am guessing that it will complain something like -- allocator closed with outstanding buffers that are not freed?